sh/folder.go

```diff
@@ -233,6 +237,12 @@ func (folder *Folder) PutObject(name string, content io.Reader) error {
 			absolutePath,
 		)
 	}
-
+	err = file.Close()
```

Comment:
Maybe switch to defer? if the above `io.Copy` fails it still leaves an opened descriptor.

Reply:
Makes sense. I'll send another PR.